extensions/wikia/SitemapXml/SpecialSitemapXmlController.class.php

```diff
@@ -73,60 +124,126 @@ private function getIso8601Timestamp( $touched ) {
 			$touched[12] . $touched[13] . 'Z';
 	}
 
-	private function getNamespaces() {
+	private function getOtherNamespaces() {
 		$nsIds = array_unique( $this->wg->ContLang->getNamespaceIds() );
+		return array_filter( $nsIds, [ $this, 'isOtherNamespace' ] );
+	}
 
-		return array_filter( $nsIds, function ( $id ) {
-			// Exclude negative namespaces
-			if ( $id < 0 ) {
-				return false;
-			}
+	private function generateSitemapPage( array $nses, $page ) {
+		$out = '<?xml version="1.0" encoding="UTF-8"?>' . PHP_EOL;
+		$out .= '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">' . PHP_EOL;
+		$out .= '<!-- Namespaces: ' . implode( ', ', $nses ) . '; page: ' . $page . ' -->' . PHP_EOL;
 
-			// Exclude talk pages
-			if ( $id % 2 !== 0 ) {
-				return false;
-			}
+		$offset = self::URLS_PER_PAGE * ( $page - 1 );
+		$limit = self::URLS_PER_PAGE;
+
+		foreach ( $nses as $ns ) {
+			$priority[$ns] = self::PRIORITIES[$ns] ?? self::DEFAULT_PRIORITY;
+			$title = Title::newFromText( '$1', $ns );
+			$urlPrefix[$ns] = str_replace( '$1', '', $title->getFullURL() );
+		}
+
+		foreach ( $this->model->getItems( $nses, $offset, $limit ) as $page ) {
+			$encodedTitle = wfUrlencode( str_replace( ' ', '_', $page->page_title ) );
+			$lastmod = $this->getIso8601Timestamp( $page->page_touched );
+
+			$out .= '<url>' . PHP_EOL;
+			$out .= '<loc>' . $urlPrefix[$page->page_namespace] . $encodedTitle . '</loc>' . PHP_EOL;
+			$out .= '<lastmod>' . $lastmod . '</lastmod>' . PHP_EOL;
+			$out .= '<priority>' . $priority[$page->page_namespace] . '</priority>' . PHP_EOL;
+			$out .= '</url>' . PHP_EOL;
+		}
+
+		$out .= '</urlset>' . PHP_EOL;
+		return $out;
+	}
+
+	private function generateSitemapIndex() {
+		$out = '<?xml version="1.0" encoding="UTF-8"?>' . PHP_EOL;
+		$out .= '<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">' . PHP_EOL;
 
-			// Excluded namespaces
-			if ( in_array( $id, self::EXCLUDED_NAMESPACES ) ) {
-				return false;
+		$baseUrl = $this->wg->Server;
+
+		foreach ( self::SEPARATE_SITEMAPS as $ns ) {
+			$ns_pages = $this->model->getPageNumber( [ $ns ], self::URLS_PER_PAGE );
+			for ( $page = 1; $page <= $ns_pages; $page++ ) {
+				$url = $baseUrl . '/sitemap-newsitemapxml-NS_' . $ns . '-p' . $page . '.xml';
+				$out .= '<sitemap><loc>' . $url . '</loc></sitemap>' . PHP_EOL;
 			}
+		}
+
+		$other_pages = $this->model->getPageNumber( $this->getOtherNamespaces(), self::URLS_PER_PAGE );
+
+		for ( $page = 1; $page <= $other_pages; $page++ ) {
+			$url = $baseUrl . '/sitemap-newsitemapxml-others-p' . $page . '.xml';
+			$out .= '<sitemap><loc>' . $url . '</loc></sitemap>' . PHP_EOL;
+		}
+
+		$out .= '</sitemapindex>' . PHP_EOL;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$out". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)